src/element-stub.js

```diff
@@ -15,33 +15,5 @@
  */
 
 import {BaseElement} from './base-element';
-import {devAssert} from './log';
 
-/** @type {!Array} */
-export const stubbedElements = [];
-
-export class ElementStub extends BaseElement {
-  /** @param {!AmpElement} element */
-  constructor(element) {
-    super(element);
-    stubbedElements.push(this);
-  }
-
-  /** @override */
-  getLayoutPriority() {
-    return devAssert(0, 'Cannot get priority of stubbed element');
-  }
-
-  /** @override */
-  isLayoutSupported(unusedLayout) {
-    // Always returns true and will eventually call this method on the actual
-    // element.
-    return true;
-  }
-
-  /** @override */
-  reconstructWhenReparented() {
-    // No real state so no reason to reconstruct.
-    return false;
-  }
-}
+export class ElementStub extends BaseElement {}
```

Comment:
I wonder if we could simplify the upgrade code if we make `ElementStub` a build-deferred extension?

Reply:
I tried that way. But I actually in the end liked the idea of not allocating any new object at all. And a bunch of `if (this.impl_) {}` I ended up not hating too much. But I can be convinced otherwise.